ipykernel/kernelapp.py

```diff
@@ -323,9 +323,13 @@ def init_io(self):
         if self.outstream_class:
             outstream_factory = import_item(str(self.outstream_class))
             sys.stdout.flush()
-            sys.stdout = outstream_factory(self.session, self.iopub_thread, u'stdout')
+            sys.stdout = outstream_factory(self.session, self.iopub_thread,
+                                           u'stdout',
+                                           echo=sys.__stdout__)
```

Comment:
This shouldn't be the default behavior. The default stdout/stderr should only be redirected by default.

Reply:
@minrk I figured as much. How do you think this would be best configured. A property on the kernel class or a property on the kernel app?

Reply:
Updated to be controlled by a ```quiet``` attribute. Similar to no_stdout.